app/models/api_key.rb

```diff
@@ -52,6 +54,14 @@ def create_bot_user
     end
   end
 
+  def delete_bot_user
+    if bot_user.present? && bot_user.owns_media?
+      bot_user.update(api_key_id: nil)
+    else
+      bot_user.destroy
```

Comment:
Just a note that I would suggest a test for this case but I see that the coverage is still 100% so it's already covered by some other test.